pkg/controller/cainjector/controller.go

```diff
@@ -145,6 +146,11 @@ func (r *genericInjectReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 	}
 	log = logf.WithResource(r.log, metaObj)
 
+	if !metaObj.GetDeletionTimestamp().IsZero() {
+		log.V(logf.TraceLevel).Info("ignoring", "reason", "object has a non-zero deletion timestamp")
+		return ctrl.Result{}, nil
+	}
```

Comment:
I added this because I suspected that tests were failing due to still-deleting resources from previous tests.
It did not solve those failures, but this is still worth keeping to prevent the controller needlessly attempting to update injectables that have finalizers and which are being deleted.

Reply:
Isn't this the other way around? I thought the deletion timestamp is only set if the object is being deleted